# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCRuntime.h
# Commit: 453aa2faad97
# Full Hash: 453aa2faad97ce68b33bef75cd232e37d759556e
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-29 09:50:10
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 4: Remove option to set maximum mark stack capacity in release builds r=sfink
#   
#   Currently we have a GC parameter that allows setting a maximum mark stack
#   capacity. This is only ever used by test code, not in the browser. This
#   requires extra unnecessary work in release builds if we move to a different
# ==============================================================================

diff -r d33708e4f8ec -r 453aa2faad97 js/src/gc/GCRuntime.h
--- a/js/src/gc/GCRuntime.h	Fri Oct 28 15:17:43 2022 +0000
+++ b/js/src/gc/GCRuntime.h	Fri Oct 28 15:17:43 2022 +0000
@@ -288,7 +288,6 @@
 
   [[nodiscard]] bool addRoot(Value* vp, const char* name);
   void removeRoot(Value* vp);
-  void setMarkStackLimit(size_t limit, AutoLockGC& lock);
 
   [[nodiscard]] bool setParameter(JSGCParamKey key, uint32_t value);
   [[nodiscard]] bool setParameter(JSGCParamKey key, uint32_t value,
@@ -394,6 +393,7 @@
   bool selectForMarking(JSObject* object);
   void clearSelectedForMarking();
   void setDeterministic(bool enable);
+  void setMarkStackLimit(size_t limit, AutoLockGC& lock);
 #endif
 
   uint64_t nextCellUniqueId() {